src/flint/types/fmpz_mod_poly.pyx

```diff
@@ -0,0 +1,91 @@
+from flint.flintlib.fmpz_mod_poly cimport *
+from flint.flintlib.fmpz cimport(
+    fmpz_equal,
+)
+from flint.types.fmpz cimport fmpz
+from flint.types.fmpz_mod cimport fmpz_mod_ctx, fmpz_mod
+from flint.types.fmpz_poly cimport any_as_fmpz_poly, fmpz_poly
+
+from flint.flint_base.flint_base cimport flint_poly
+from flint.utils.typecheck cimport typecheck
+
+cdef class fmpz_mod_poly_ctx(fmpz_mod_ctx):
+    """
+    NOTE:
+
+    Technically this could just be the same as `fmpz_mod_ctx`,
+    however, the usage of fmpz_mod_ctx allows the creation of 
+    `fmpz_mod` types by calling the context class. For symmetry
+    we allow this to be the case here.
+    """
+    def __eq__(self, other):
+        # Most often, we expect both `fmpz_mod_poly` to be pointing 
+        # to the same ctx, so this seems the fastest way to check
+        if self is other:
+            return True
+        
+        # If they're not the same object in memory, they may have the
+        # same modulus, which is good enough
+        if typecheck(other, fmpz_mod_poly_ctx):
+            return fmpz_equal(self.val.n, (<fmpz_mod_poly_ctx>other).val.n)
+        return False
+
+    def __str__(self):
+        return f"Context for fmpz_mod_poly with modulus: {self.modulus()}"
+
+    def __repr__(self):
+        return f"fmpz_mod_poly_ctx({self.modulus()})"
+
+    def __call__(self, val):
+        return fmpz_mod_poly(val, self)
+
+
+cdef class fmpz_mod_poly(flint_poly):
+    """
+    """
+    def __cinit__(self):
+        fmpz_mod_poly_init(self.val, self.ctx.val)
+
+    def __dealloc__(self):
+        fmpz_mod_poly_clear(self.val, self.ctx.val)
+
+    def __init__(self, val, ctx):
+        if not typecheck(ctx, fmpz_mod_ctx):
+            raise TypeError
+        self.ctx = ctx
+
+        val = any_as_fmpz_poly(val)
+        if val is NotImplemented:
+            raise TypeError
+
+        fmpz_mod_poly_set_fmpz_poly(
+            self.val, (<fmpz_poly>val).val, self.ctx.val
+        )
+
+    def __getitem__(self, long i):
+        cdef fmpz x
+        x = fmpz()
+        if i < 0:
+            return x
+        fmpz_mod_poly_get_coeff_fmpz(x.val, self.val, i, self.ctx.val)
```

Comment:
Should this not return an `fmpz_mod`?

Reply:
Yes, and with the above change this is also easy :)